### DISCUSSION TITLE
Development notes

### DISCUSSION THREAD

**Discussion by shred86** [humbertogontijo/homeassistant-roborock#89] [General]
@humbertogontijo I'm creating this discussion just to document stuff I'm seeing as I'm writing tests. Figured it would be a good way to just share thoughts and discuss development related stuff.

The first thing I noted is the camera uses just the device uid `unique_id`. While this likely won't matter since a user should only have one camera entity for the map, I do wonder if it would be smart to just prefix it with something like you did for the other entities. I noticed the vacuum itself uses just the device uid as well for it's `unique_id` but HA isn't throwing any errors, so I'm assuming they might be specific to each entity domain.

**Comment by mvdwetering** [comment]
Uniqueids just need to be unique per platform within an integration

See `unique_id` entry on https://developers.home-assistant.io/docs/core/entity#generic-properties and "Learn more" link for more info.

>A unique identifier for this entity. Needs to be unique within a platform (ie light.hue). Should not be configurable by the user or be changeable. [Learn more.](https://developers.home-assistant.io/docs/entity_registry_index#unique-id-requirements)
